modules/hostap/src/supp_main.c

```diff
@@ -306,8 +302,15 @@ static void iface_event_handler(struct net_mgmt_event_callback *cb,
 							uint32_t mgmt_event, struct net_if *iface)
 {
 	const char *ifname = iface->if_dev->dev->name;
+	const struct device *wifi_dev = NULL;
+
+	wifi_dev =  DEVICE_DT_GET(DT_CHOSEN(zephyr_wifi));
+	if (!wifi_dev) {
+		wpa_printf(MSG_ERROR, "Device instance not found");
+		return;
+	}
 
-	if (strncmp(ifname, IFACE_MATCHING_PREFIX, sizeof(IFACE_MATCHING_PREFIX) - 1) != 0) {
+	if (strncmp(ifname, wifi_dev->name, sizeof(wifi_dev->name) - 1) != 0) {
```

Comment:
This won't work, as it needs to match multiple interfaces.

Reply:
Introduced a Kconfig option for prefix string.